server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -40,6 +41,13 @@
  */
 public abstract class BaseRestHandler implements RestHandler {
 
+    /**
+     * Parameter that controls whether certain REST apis should include type names in their requests or responses.
+     * Note: This parameter is only available through compatible rest api.
```

Comment:
might want to reference the V_7 constant in the java doc ... i think the build will fail if you reference something that is not there and will help to find this when it is time to re-remove.